libs/design-system/dialog/src/lib/notice/notice.component.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { TestBed } from '@angular/core/testing';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { render, screen } from '@testing-library/angular';
+import { DialogService } from '../../dialog.service';
+import { NoticeComponent } from './notice.component';
+import userEvent from '@testing-library/user-event';
+
+describe('DialogComponent', () => {
+  let service: DialogService;
+  const TITLE = 'Test Title';
+  const MESSAGE = 'Test Message';
+  beforeEach(async () => {
+    await render(NoticeComponent, {
+      providers: [
+        { provide: MAT_DIALOG_DATA, useValue: {} },
+        { provide: MatDialogRef, useValue: {} },
+      ],
+    });
+
+    service = TestBed.inject(DialogService);
+  });
+
+  it('Title and Message should be present when dialog is opened', async () => {
+    service.openNotice(TITLE, MESSAGE);
+
+    const title = await screen.findByText(TITLE);
+    expect(title).toBeInTheDocument();
+
+    const message = await screen.findByText(MESSAGE);
+    expect(message).toBeInTheDocument();
+
+    const dismiss = await screen.findByRole('button', { name: 'Dismiss' });
+    expect(dismiss).toBeInTheDocument();
+  });
+
+  it('Action button should be visible', async () => {
+    service.openNotice(TITLE, MESSAGE, {
+      label: 'Action Button',
+      callback: () => {
+        ('');
```

Comment:
Woot! Better practice it to use jest.fn() when creating empty functions. By doing so you can also test that the passed function is called when the action button is clicked.
```ts
const callback = jest.fn();
service.openNotice(..., {
   ...,
   callback
});

...

await user.click(dismiss)

expect(callback).toHaveBeenCalledTimes(1);
```